src/SWP/Bundle/ContentBundle/Manager/MediaManager.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+/**
+ * This file is part of the Superdesk Web Publisher Content Bundle.
+ *
+ * Copyright 2016 Sourcefabric z.ú. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2016 Sourcefabric z.ú.
+ * @license http://www.superdesk.org/license
+ */
+namespace SWP\Bundle\ContentBundle\Manager;
+
+use SWP\Bundle\ContentBundle\Model\FileInterface;
+use SWP\Component\MultiTenancy\Context\TenantContextInterface;
+use SWP\Component\MultiTenancy\PathBuilder\TenantAwarePathBuilder;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+use League\Flysystem\Filesystem;
+use Doctrine\ODM\PHPCR\DocumentManager;
+use SWP\Bundle\ContentBundle\Doctrine\ODM\PHPCR\File;
+use SWP\Bundle\ContentBundle\Doctrine\ODM\PHPCR\Image;
+use Symfony\Component\Routing\RequestContext;
+use Symfony\Component\Routing\RouterInterface;
+
+class MediaManager implements MediaManagerInterface
+{
+    /**
+     * @var TenantAwarePathBuilder
+     */
+    protected $pathBuilder;
+
+    /**
+     * @var Filesystem
+     */
+    protected $filesystem;
+
+    /**
+     * @var DocumentManager
+     */
+    protected $objectManager;
+
+    /**
+     * @var string
+     */
+    protected $mediaBasepath;
+
+    /**
+     * MediaManager constructor.
+     *
+     * @param TenantAwarePathBuilder $pathBuilder
+     * @param Filesystem             $filesystem
+     * @param DocumentManager        $objectManager
+     * @param RouterInterface        $router
+     * @param TenantContextInterface $tenantContext
+     * @param string                 $mediaBasepath
+     */
+    public function __construct(
+        TenantAwarePathBuilder $pathBuilder,
+        Filesystem $filesystem,
+        DocumentManager $objectManager,
+        RouterInterface $router,
+        TenantContextInterface $tenantContext,
+        $mediaBasepath
+    ) {
+        $this->pathBuilder = $pathBuilder;
+        $this->filesystem = $filesystem;
+        $this->objectManager = $objectManager;
+        $this->mediaBasepath = $mediaBasepath;
+        $this->router = $router;
+        $this->tenantContext = $tenantContext;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handleUploadedFile(UploadedFile $uploadedFile, $mediaId)
+    {
+        $this->saveFile($uploadedFile, $mediaId);
+
+        $media = $this->getProperObject($uploadedFile);
+        $media->setParentDocument($this->objectManager->find(null, $this->pathBuilder->build($this->mediaBasepath)));
```

Comment:
this will work only for PHPCR as far as I see because you use File and Media classes for PHPCR implementation explicitly, this class should be moved under PHPCR namespace or you can create a factory and use it here instead - you should then have a generic manager.